OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -17,13 +19,36 @@ public class HealthInfo : ITraitInfo, UsesInit<HealthInit>
 	{
 		[Desc("HitPoints")]
 		public readonly int HP = 0;
-
-		[Desc("Physical size of the unit used for damage calculations. Impacts within this radius apply full damage.")]
-		public readonly WDist Radius = new WDist(426);
-
 		[Desc("Trigger interfaces such as AnnounceOnKill?")]
 		public readonly bool NotifyAppliedDamage = true;
 
+		[FieldLoader.LoadUsing("LoadShape")]
+		public readonly IHitShape Shape;
+
+		static object LoadShape(MiniYaml yaml)
+		{
+			var nodes = yaml.ToDictionary();
+
+			string shape = "Circle";
+			if (nodes.ContainsKey("Shape"))
+				shape = FieldLoader.GetValue<string>("Shape", nodes["Shape"].Value);
+
+			int radius = 426;
+			if (nodes.ContainsKey("Radius"))
```

Comment:
Is this what it is supposed to parse?
```Yaml
MCV:
    Health:
        HP: 600
        Shape: Capsule
            Radius: 426
            Vertices: 0, -512, 0, 512
```
If so, then `nodes = nodes["Shape"].Value`.